<!-- gbx_cropping/launch/gbx_cropping.launch -->
<launch>
    <arg name="camera_topic" default="/hk_camera/image_raw" doc="输入图像话题名称"/>
    <arg name="output_dir" default="/home/lsy/gbx_cropping_ws/src/gbx_cropping/cropped_images" doc="裁剪图像保存路径"/>

<!--    <node pkg="nodelet" type="nodelet" name="gbx_cropping_nodelet_manager" args="manager" output="screen"/>-->

    <node pkg="nodelet" type="nodelet" name="gbx_cropping_nodelet"
          args="load gbx_cropping/GBXCroppingNodelet vision_nodelet" output="screen">

        <remap from="/hk_camera/image_raw" to="$(arg camera_topic)"/>
        <param name="block_size" value="11"/>
        <param name="C" value="2.0"/>
        <param name="blur_size" value="9"/>
        <param name="min_area" value="3000.0"/>
        <param name="max_area" value="500000.0"/>
        <param name="circularity_threshold" value="0.1"/>
<!--        <param name="output_directory" value="$(arg output_dir)"/>-->
<!--        <param name="block_size" value="11"/>-->
<!--        <param name="C" value="2.0"/>-->
<!--        <param name="min_area" value="3000.0"/>-->
<!--        <param name="max_area" value="500000.0"/>-->
<!--        <param name="circularity_threshold" value="0.1"/>-->
<!--        <param name="close_kernel_size" value="2"/>-->
<!--        <param name="close_iterations" value="3"/>-->
<!--        <param name="close_operation" value="0"/>-->
<!--        <param name="dilate_kernel_size" value="2"/>-->
<!--        <param name="dilate_iterations" value="3"/>-->
<!--        <param name="dilate_operation" value="0"/>-->
    </node>

<!--    <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure" output="screen"/>-->

<!--    <node pkg="rqt_image_view" type="rqt_image_view" name="image_view_annotated" output="screen">-->
<!--        <remap from="image" to="annotated_image"/>-->
<!--    </node>-->

<!--    <node pkg="rqt_image_view" type="rqt_image_view" name="image_view_stitched" output="screen">-->
<!--        <remap from="image" to="stitched_image"/>-->
<!--    </node>-->
</launch>